File: base/src/main/java/com/fubao/baselibrary/utils/ActivityManager.java
package com.fubao.baselibrary.utils;

import android.app.Activity;

import java.lang.ref.WeakReference;
import java.util.ArrayList;

/**
 * 所有Activity的管理类，可以将所有Activity统一保存起来，以便于退出应用的时候kill所有Activity。
 */
public class ActivityManager {

    private static final ArrayList<WeakReference<Activity>> activityList = new ArrayList<>();

    public static void add(WeakReference<Activity> activityWeakReference) {
        activityList.add(activityWeakReference);
    }

    public static void remove(WeakReference<Activity> activityWeakReference) {
        activityList.remove(activityWeakReference);
    }

    public static void finishAll() {
        for (WeakReference<Activity> activityWeakReference : activityList) {
            Activity activity = activityWeakReference.get();
            if (activity != null && !activity.isFinishing()) {
                activity.finish();
            }
        }
        activityList.clear();
    }
}

File: base/src/main/java/com/zqlq/common/utils/ActivityManager.kt
package com.zqlq.common.utils

import android.app.Activity
import java.lang.ref.WeakReference

/**
 * 所有Activity的管理类，可以将所有Activity统一保存起来，以便于退出应用的时候kill所有Activity。
 */
object ActivityManager {

    private val activityList = ArrayList<WeakReference<Activity>>()

    @JvmStatic
    fun add(activityWeakReference: WeakReference<Activity>) {
        activityList.add(activityWeakReference)
    }

    @JvmStatic
    fun remove(activityWeakReference: WeakReference<Activity>) {
        activityList.remove(activityWeakReference)
    }

    @JvmStatic
    fun finishAll() {
        for (activityWeakReference in activityList) {
            val activity = activityWeakReference.get()
            if (activity != null && !activity.isFinishing) {
                activity.finish()
            }
        }
        activityList.clear()
    }
}
